compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/lower/AbstractComposeLowering.kt

```diff
@@ -904,63 +898,86 @@ abstract class AbstractComposeLowering(
     }
 
-    private fun IrClass.makeStabilityFieldJvm(): IrField {
+    private fun IrField.stabilityFieldValue(): IrGetField =
+        IrGetFieldImpl(
+            UNDEFINED_OFFSET,
+            UNDEFINED_OFFSET,
+            this.symbol,
+            this.type
+        )
+
+    private fun IrClass.runtimeStabilityValue(): IrExpression {
+        return if (context.platform.isJvm()) {
+            val stableField = this.makeStabilityFieldJvm()
+            stableField.stabilityFieldValue()
+        } else {
+            val stableProp = this.makeStabilityProp()
+            irCall(stableProp.getter!!, SYNTHETIC_OFFSET, SYNTHETIC_OFFSET)
+        }
+    }
+
+    private fun makeStabilityField(fieldName: Name): IrField {
         return context.irFactory.buildField {
             startOffset = SYNTHETIC_OFFSET
             endOffset = SYNTHETIC_OFFSET
-            name = KtxNameConventions.STABILITY_FLAG
+            name = fieldName
             isStatic = true
             isFinal = true
             type = context.irBuiltIns.intType
             visibility = DescriptorVisibilities.PUBLIC
-        }.also { stabilityField ->
+        }
+    }
+
+    private fun IrClass.makeStabilityFieldJvm(): IrField {
+        return makeStabilityField(KtxNameConventions.STABILITY_FLAG).also { stabilityField ->
             stabilityField.parent = this@makeStabilityFieldJvm
         }
     }
 
-    private fun IrClass.makeStabilityFieldNonJvm(): IrField {
-        val stabilityFieldName = this.uniqueStabilityFieldName()
+    private fun IrClass.makeStabilityProp(): IrProperty {
         val fieldParent = this.getPackageFragment()
 
-        return context.irFactory.buildField {
-            startOffset = SYNTHETIC_OFFSET
-            endOffset = SYNTHETIC_OFFSET
-            name = stabilityFieldName
-            isStatic = true
-            isFinal = true
-            type = context.irBuiltIns.intType
-            visibility = DescriptorVisibilities.PUBLIC
-        }.let { stabilityField ->
-            stabilityField.parent = fieldParent
-            makeStabilityProp(stabilityField, fieldParent)
-        }.backingField!!
-    }
-
-    private fun IrClass.makeStabilityProp(
-        stabilityField: IrField,
-        fieldParent: IrDeclarationContainer
-    ): IrProperty {
         val propName = this@makeStabilityProp.uniqueStabilityPropertyName()
         val existingProp = fieldParent.declarations.firstOrNull {
             it is IrProperty && it.name == propName
         } as? IrProperty
         if (existingProp != null) {
             return existingProp
         }
+
+        val stabilityField = makeStabilityField(this.uniqueStabilityFieldName()).also {
+            it.parent = fieldParent
+        }
         return context.irFactory.buildProperty {
             startOffset = SYNTHETIC_OFFSET
             endOffset = SYNTHETIC_OFFSET
             name = propName
             visibility = DescriptorVisibilities.PUBLIC
-            isConst = true
```

Comment:
wdyt, if we keep this line only for 1.9.2x versions, but get rid of it for 2.0.x? 

I guess it might help preserve the compatibility for more cases.